src/components/Dialogs/SeeTaskDialog.vue

```diff
@@ -0,0 +1,132 @@
+            <circle cx="2.308" cy="2.308" r="2.308" />
+            <circle cx="2.308" cy="10" r="2.308" />
+            <circle cx="2.308" cy="17.692" r="2.308" />
+          </g>
+        </svg>
+      </the-button>
+    </template>
+
+    <template #main-content>
+      <div class="flex flex-col gap-6 relative">
+        <more-options
+          :condition="areTaskOptionsShown"
+          element="task"
+          :toggleOptions="(e: Event) => callMoreOptionsFn(e, toggleOptions)"
+          :closeMoreOptions="(e: Event) => callMoreOptionsFn(e, closeOptions)"
+          :showEditForm="showEditForm"
+          :showDeleteForm="showDeleteForm"
+          :closeSeeTask="closeSeeTask"
+        />
+
+        <p class="text-gray-400 text-xs min-[350px]:text-sm">
+          {{ boardProperties.description }}
+        </p>
+
+        <div>
+          <p class="mb-4 text-xs text-gray-400 dark:text-white">
+            Subtasks (
+            {{ returnNumberOfCompletedSubtasks(boardProperties.subtasks) }}
+            of
+            {{ boardProperties.subtasks.length }}
+            )
+          </p>
+          <div
+            v-for="subtask in boardProperties.subtasks"
+            :key="subtask.title"
+            class="subtask"
+          >
+            <label class="flex items-center gap-4 px-1 cursor-pointer">
+              <input
+                type="checkbox"
+                :checked="subtask.isCompleted"
+                class="checkbox peer"
+              />
+              <span
+                class="text-xs peer-checked:line-through peer-checked:opacity-50"
+              >
+                {{ subtask.title }}
+              </span>
+            </label>
+          </div>
+        </div>
+
+        <div>
+          <p class="mb-2 text-xs text-gray-400 dark:text-white">
+            Current Status
+          </p>
+          <v-select
+            :options="statusItems"
+            :searchable="false"
+            placeholder="Doing"
+          ></v-select>
+        </div>
+      </div>
+    </template>
+  </dialogs-template>
+</template>
+
+<script setup lang="ts">
+import DialogsTemplate from './DialogsTemplate.vue'
+import MoreOptions from '../shared/MoreOptions.vue'
+import TheButton from '../shared/TheButton.vue'
+import moreOptionsPopup from '../../composables/moreOptionsPopup'
+import { returnBoardProperties } from '../../composables/boardProperties'
+import { returnNumberOfCompletedSubtasks } from '../../composables/completedTasks'
+import { ref, Ref } from 'vue'
+
+defineProps<{
+  showEditForm: () => void
+  showDeleteForm: () => void
+  closeSeeTask: () => void
+}>()
+
+const boardProperties = returnBoardProperties()
+const statusItems = boardProperties.columns.map((column) => column.name)
+const areTaskOptionsShown = ref(false)
+
+const { toggleOptions, closeOptions } = moreOptionsPopup
+const callMoreOptionsFn = (
+  e: Event,
+  cb: (e: Event, conditionToChange: Ref<boolean>) => void
```

Comment:
> Why do you pass a ref instead of its value?

Not sure (when doing that I just tried different things until something worked) but I can see now, the function doesn't work when I pass `areTaskOptionsShown.value` instead of `areTaskOptionsShown`. I believe, it's because the callback is effectively being called in the template, where `areTaskOptionsShown.value` doesn't exist. But I may be wrong on that. Still, passing a ref to that seems to be the only way to go.


> Avoid data mutation outside it's component

You mean, I shouldn't have those helper functions? I did this because I actually need them in two separate component and I didn't want to violate this DRY thing you mentioned in one of your following comments...